packages/sdk-components-react/src/time.tsx

```diff
@@ -0,0 +1,371 @@
+  "LR",
+  "LY",
+  "LI",
+  "LT",
+  "LU",
+  "MG",
+  "MW",
+  "MY",
+  "MV",
+  "ML",
+  "MT",
+  "MH",
+  "MR",
+  "MU",
+  "MX",
+  "FM",
+  "MD",
+  "MC",
+  "MN",
+  "ME",
+  "MA",
+  "MZ",
+  "MM",
+  "NA",
+  "NR",
+  "NP",
+  "NL",
+  "NZ",
+  "NI",
+  "NE",
+  "NG",
+  "NO",
+  "OM",
+  "PK",
+  "PW",
+  "PA",
+  "PG",
+  "PY",
+  "PE",
+  "PH",
+  "PL",
+  "PT",
+  "QA",
+  "RO",
+  "RU",
+  "RW",
+  "KN",
+  "LC",
+  "VC",
+  "WS",
+  "SM",
+  "ST",
+  "SA",
+  "SN",
+  "RS",
+  "SC",
+  "SL",
+  "SG",
+  "SK",
+  "SI",
+  "SB",
+  "SO",
+  "ZA",
+  "SS",
+  "ES",
+  "LK",
+  "SD",
+  "SR",
+  "SE",
+  "CH",
+  "SY",
+  "TW",
+  "TJ",
+  "TZ",
+  "TH",
+  "TL",
+  "TG",
+  "TO",
+  "TT",
+  "TN",
+  "TR",
+  "TM",
+  "TV",
+  "UG",
+  "UA",
+  "AE",
+  "GB",
+  "US",
+  "UY",
+  "UZ",
+  "VU",
+  "VA",
+  "VE",
+  "VN",
+  "YE",
+  "ZM",
+  "ZW",
+] as const;
+
+type Language = (typeof languages)[number];
+type Country = (typeof countries)[number];
+type DateStyle = Intl.DateTimeFormatOptions["dateStyle"] | "none";
+type TimeStyle = Intl.DateTimeFormatOptions["timeStyle"] | "none";
+
+export type TimeProps = {
+  datetime?: string;
+  language?: Language;
+  country?: Country;
+  dateStyle?: DateStyle;
+  timeStyle?: TimeStyle;
+};
+
+const INITIAL_DATE_STRING = "dateTime attribute is not set";
+
+const DEFAULT_LANGUAGE = "en";
+const DEFAULT_COUNTRY = "GB";
+const DEFAULT_DATE_STYLE = "medium";
+const DEFAULT_TIME_STYLE = "none";
+
+const languageOrDefault = (language: unknown): Language => {
+  return languages.includes(language as Language)
+    ? (language as Language)
+    : DEFAULT_LANGUAGE;
+};
+
+const countryOrDefault = (country: unknown): Country => {
+  return countries.includes(country as Country)
+    ? (country as Country)
+    : DEFAULT_COUNTRY;
+};
+
+const dateStyleOrUndefined = (
+  value: unknown
+): Intl.DateTimeFormatOptions["dateStyle"] => {
+  if (["full", "long", "medium", "short"].includes(value as string)) {
+    return value as Intl.DateTimeFormatOptions["dateStyle"];
+  }
+  return undefined;
+};
+
+const timeStyleOrUndefined = (
+  value: unknown
+): Intl.DateTimeFormatOptions["timeStyle"] => {
+  if (["full", "long", "medium", "short"].includes(value as string)) {
+    return value as Intl.DateTimeFormatOptions["timeStyle"];
+  }
+  return undefined;
+};
+
+export const Time = React.forwardRef<React.ElementRef<"time">, TimeProps>(
+  (
+    {
+      language = DEFAULT_LANGUAGE,
+      country = DEFAULT_COUNTRY,
+      dateStyle = DEFAULT_DATE_STYLE,
+      timeStyle = DEFAULT_TIME_STYLE,
+      datetime = INITIAL_DATE_STRING,
+      ...props
+    },
+    ref
+  ) => {
+    const locale = `${languageOrDefault(language)}-${countryOrDefault(
+      country
+    )}`;
+
+    const options: Intl.DateTimeFormatOptions = {
+      dateStyle: dateStyleOrUndefined(dateStyle),
+      timeStyle: timeStyleOrUndefined(timeStyle),
+    };
+
+    const datetimeString = datetime === null ? "" : datetime.toString();
```

Comment:
why null?

Reply:
I would better do String(datetime) as input can be undefined as well

Reply:
Null is not overwritten with = DEFAULT

Reply:
🤷 render "null" then

Reply:
new Date(null).getTime() Valid ;-) so need to be excluded

Reply:
new Date("").getTime() gives NaN

Reply:
shite

Reply:
Renamed "" to INVALID_DATE_STRING